bin/ruby-build

```diff
@@ -371,6 +371,10 @@ fetch_tarball() {
   local checksum
   local extracted_dir
 
+  if is_ruby_package "$1" && [ -n "$RUBY_BUILD_TARBALL_OVERRIDE" ]; then
```

Comment:
The reason we need the is_ruby_package is because fetch_tarball is also used for non-ruby packages like openssl and overriding both URLs with the same URL would make no sense.